.github/workflows/ci.yml

```diff
@@ -0,0 +1,27 @@
+name: Java CI
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    - name: Set up JDK 21
+      uses: actions/setup-java@v2
+      with:
+        distribution: 'adopt'
+        java-version: '21'
+
+    - name: Grant execute permission for setup script
+      run: chmod +x setup.sh
```

Comment:
Please update setup.sh to have execution permission, and let's push . No reason to give permission before each execution.

Reply:
@relentless-pursuit , please remove this line and lets merge this PR.

Reply:
we can comment out 
```
- name: Run setup script
      run: ./setup.sh
```
for now 
and uncomment it when we be ready execution of checktyle repo

Reply:
done